### PR TITLE
spread out docker layers to make image pulling faster

### PR DESCRIPTION
**Pull request by dimberman**

### CODE REVIEW

**Comment by dimberman** [comment]
Before:
```584f1244e4b4        About an hour ago   /bin/sh -c #(nop)  ENTRYPOINT ["tini" "--" "…   0B
<missing>           About an hour ago   /bin/sh -c #(nop)  EXPOSE 5555 8080 8793        0B
<missing>           About an hour ago   /bin/sh -c #(nop) WORKDIR /usr/local/airflow    0B
<missing>           About an hour ago   |4 BUILD_NUMBER=1302 ORG=astronomer SUBMODUL…   0B
<missing>           About an hour ago   /bin/sh -c #(nop) COPY file:242f5b9ada91800e…   461B
<missing>           About an hour ago   /bin/sh -c #(nop) COPY file:357176992ee63516…   1.28kB
<missing>           About an hour ago   |4 BUILD_NUMBER=1302 ORG=astronomer SUBMODUL…   0B
<missing>           About an hour ago   |4 BUILD_NUMBER=1302 ORG=astronomer SUBMODUL…   584MB
<missing>           About an hour ago   |4 BUILD_NUMBER=1302 ORG=astronomer SUBMODUL…   286B
<missing>           About an hour ago   |4 BUILD_NUMBER=1302 ORG=astronomer SUBMODUL…   4.85kB
<missing>           About an hour ago   /bin/sh -c #(nop)  ENV ASTRONOMER_GROUP=astro   0B
<missing>           About an hour ago   /bin/sh -c #(nop)  ENV ASTRONOMER_USER=astro    0B
<missing>           About an hour ago   /bin/sh -c #(nop)  ARG ASTRONOMER_GROUP=astro   0B
<missing>           About an hour ago   /bin/sh -c #(nop)  ARG ASTRONOMER_USER=astro    0B
<missing>           About an hour ago   /bin/sh -c #(nop)  ENV AIRFLOW_GPL_UNIDECODE…   0B
<missing>           About an hour ago   /bin/sh -c #(nop)  ENV PYTHONPATH=/usr/local…   0B
<missing>           About an hour ago   /bin/sh -c #(nop)  ENV PYMSSQL_BUILD_WITH_BU…   0B
<missing>           About an hour ago   /bin/sh -c #(nop)  ENV AIRFLOW_HOME=/usr/loc…   0B
<missing>           About an hour ago   /bin/sh -c #(nop)  ENV AIRFLOW_MODULE=git+ht…   0B
<missing>           About an hour ago   /bin/sh -c #(nop)  ENV AIRFLOW_REPOSITORY=ht…   0B
<missing>           About an hour ago   /bin/sh -c #(nop)  ARG SUBMODULES=all, stats…   0B
<missing>           About an hour ago   /bin/sh -c #(nop)  ARG VERSION=v1-10-5          0B
<missing>           About an hour ago   /bin/sh -c #(nop)  ARG ORG=astronomer           0B
<missing>           About an hour ago   /bin/sh -c #(nop)  LABEL io.astronomer.docke…   0B
<missing>           About an hour ago   /bin/sh -c #(nop)  LABEL io.astronomer.docke…   0B
<missing>           About an hour ago   /bin/sh -c #(nop)  LABEL io.astronomer.docke…   0B
<missing>           About an hour ago   /bin/sh -c #(nop)  LABEL io.astronomer.docke…   0B
<missing>           About an hour ago   /bin/sh -c #(nop)  LABEL io.astronomer.docke…   0B```

**Comment by dimberman** [comment]
after:
```f891d6d9aa38        53 seconds ago       /bin/sh -c #(nop)  ENTRYPOINT ["tini" "--" "…   0B
460f46fed1ba        54 seconds ago       /bin/sh -c #(nop)  EXPOSE 5555 8080 8793        0B
3fce06887c2f        54 seconds ago       /bin/sh -c #(nop) WORKDIR /usr/local/airflow    0B
9f47edd491b9        54 seconds ago       |4 BUILD_NUMBER=1 ORG=astronomer SUBMODULES=…   0B
e814eb032337        56 seconds ago       /bin/sh -c #(nop) COPY file:242f5b9ada91800e…   461B
08523c963ad6        56 seconds ago       /bin/sh -c #(nop) COPY file:357176992ee63516…   1.28kB
05684e69c16a        57 seconds ago       |4 BUILD_NUMBER=1 ORG=astronomer SUBMODULES=…   0B
9fe27247d98c        58 seconds ago       |4 BUILD_NUMBER=1 ORG=astronomer SUBMODULES=…   35.1MB
b318f7872dab        About a minute ago   |4 BUILD_NUMBER=1 ORG=astronomer SUBMODULES=…   346MB
2208fdfff851        6 minutes ago        |4 BUILD_NUMBER=1 ORG=astronomer SUBMODULES=…   118MB
1a2955605d04        6 minutes ago        |4 BUILD_NUMBER=1 ORG=astronomer SUBMODULES=…   33.2MB
da5175df1e52        6 minutes ago        |4 BUILD_NUMBER=1 ORG=astronomer SUBMODULES=…   351MB
3343b690bce0        7 minutes ago        |4 BUILD_NUMBER=1 ORG=astronomer SUBMODULES=…   286B
6ecbbb26be6f        7 minutes ago        |4 BUILD_NUMBER=1 ORG=astronomer SUBMODULES=…   4.85kB
6172f5a10e97        7 minutes ago        /bin/sh -c #(nop)  ENV ASTRONOMER_GROUP=astro   0B
a88dad6ddbc0        7 minutes ago        /bin/sh -c #(nop)  ENV ASTRONOMER_USER=astro    0B
3e96c60a249a        7 minutes ago        /bin/sh -c #(nop)  ARG ASTRONOMER_GROUP=astro   0B
7a225e13dfc2        7 minutes ago        /bin/sh -c #(nop)  ARG ASTRONOMER_USER=astro    0B
359d72429e4a        7 minutes ago        /bin/sh -c #(nop)  ENV AIRFLOW_GPL_UNIDECODE…   0B
6400b456e9b6        7 minutes ago        /bin/sh -c #(nop)  ENV PYTHONPATH=/usr/local…   0B
955e3aa8373f        7 minutes ago        /bin/sh -c #(nop)  ENV PYMSSQL_BUILD_WITH_BU…   0B
87111a4ea68f        7 minutes ago        /bin/sh -c #(nop)  ENV AIRFLOW_HOME=/usr/loc…   0B
1006484792d3        7 minutes ago        /bin/sh -c #(nop)  ENV AIRFLOW_MODULE=git+ht…   0B
0e7c7f4392ee        7 minutes ago        /bin/sh -c #(nop)  ENV AIRFLOW_REPOSITORY=ht…   0B
e0d7bd30e9e2        7 minutes ago        /bin/sh -c #(nop)  ARG SUBMODULES=all, stats…   0B
7475e8b1efd2        7 minutes ago        /bin/sh -c #(nop)  ARG VERSION=v1-10-5          0B
8f575e24e3a0        7 minutes ago        /bin/sh -c #(nop)  ARG ORG=astronomer           0B
cf25c8c436a9        7 minutes ago        /bin/sh -c #(nop)  LABEL io.astronomer.docke…   0B
94cc26a7f3da        7 minutes ago        /bin/sh -c #(nop)  LABEL io.astronomer.docke…   0B
e2f4ff9a75f5        7 minutes ago        /bin/sh -c #(nop)  LABEL io.astronomer.docke…   0B
9ee6561d5927        7 minutes ago        /bin/sh -c #(nop)  LABEL io.astronomer.docke…   0B
f002f79e7a83        7 minutes ago        /bin/sh -c #(nop)  LABEL io.astronomer.docke…   0B
241352d63a6c        7 minutes ago        /bin/sh -c #(nop)  ARG BUILD_NUMBER=-1          0B
01a9324e63a6        7 minutes ago        /bin/sh -c #(nop)  LABEL maintainer=Astronom…   0B```

File: docker/airflow/1.10.5/Dockerfile [hunk 1]
```diff
@@ -85,13 +86,13 @@ RUN apk update \
...
 	&& pip3 install --no-cache-dir --upgrade pip==19.2.1 \
 	&& pip3 install --no-cache-dir --upgrade setuptools==41.0.1 \
 	&& pip3 install --no-cache-dir "${AIRFLOW_MODULE}" \
-	&& pip3 install --no-cache-dir "https://github.com/astronomer/astronomer-fab-securitymanager/releases/download/v1.2.1/astronomer_fab_security_manager-1.2.1-py3-none-any.whl" \
-	&& cd /usr/lib/python3.7/site-packages/airflow/www_rbac \
+	&& pip3 install --no-cache-dir "https://github.com/astronomer/astronomer-fab-securitymanager/releases/download/v1.2.1/astronomer_fab_security_manager-1.2.1-py3-none-any.whl"
+RUN cd /usr/lib/python3.7/site-packages/airflow/www_rbac \
 	&& npm install \
 	&& npm run build \
 	&& rm -rf node_modules \
```
Comment (ashb) [inline docker/airflow/1.10.5/Dockerfile]: If we split this command out in to a different layer then deleting the files doesn't save us any space on the pull.

Comment (ashb) [reply docker/airflow/1.10.5/Dockerfile]: Well - some from this, but the next line (`apk del .build-deps py3-numpy-dev`) doesn't save much space.

Comment (dimberman) [reply docker/airflow/1.10.5/Dockerfile]: Yeah that's true... so I guess the question is does the increased-space counteract the better distribution of layers?

Comment (schnie) [reply docker/airflow/1.10.5/Dockerfile]: Another option could be a multi-stage build, but not sure if its worth that effort or not.

**Comment by ashb** [comment]
Probably not needed with https://github.com/astronomer/houston-api/pull/156

**Pull request closed without merging**